File: Breeze.Domain/category.cs
namespace Breeze.Domain
{
    public class Category
    {
        public int Id { get; set; }
        public string userId { get; set; }
        public string name { get; set; }
        public DateTime date { get; set; }
        public decimal budget { get; set; }
        public decimal currentSpend { get; set; }
    }
}

File: Breeze.Web/app/budget/types/category.ts
import { z } from 'zod';

export interface Category {
  id: string;
  userId: string;
  budgetId: string;
  name: string;
  allocation: string;
  currentSpend: string;
  sourceType: 'MANUAL' | 'RECURRING_TEMPLATE';
  sourceTemplateId?: string | null;
  generationMonth?: string | null;
  createdAt: string;
  updatedAt: string;
}

export const categoryFormSchema = z.object({
  name: z.string().min(1, 'Name is required'),
  allocation: z.string().min(1, 'Allocation is required'),
});

export type CategoryFormData = z.infer<typeof categoryFormSchema>;
